stubs/python-decouple/decouple.pyi

```diff
@@ -0,0 +1,162 @@
+from _typeshed import Incomplete
+from collections import OrderedDict
+from collections.abc import Callable, Iterable, Sequence
+from configparser import ConfigParser
+from typing import Any, Generic, Protocol, TextIO, TypeVar, overload
+
+_T = TypeVar("_T")
+_TCsv = TypeVar("_TCsv")
+
+PYVERSION: Incomplete  # undocumented
+DEFAULT_ENCODING: str  # undocumented
+TRUE_VALUES: set[str]  # undocumented
+FALSE_VALUES: set[str]  # undocumented
+
+def read_config(parser: ConfigParser, file: TextIO) -> None: ...  # undocumented
+def strtobool(value: str | bool) -> bool: ...  # undocumented
+
+class UndefinedValueError(Exception): ...
+class Undefined: ...
+
+undefined: Undefined
+
+class Config:
+    repository: _Repository
+    def __init__(self, repository: _Repository) -> None: ...
+    @overload
+    def get(self, option: str, default: str = ...) -> str: ...
+    @overload
+    def get(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def get(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def __call__(self, option: str, default: str = ...) -> str: ...
+    @overload
+    def __call__(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def __call__(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
```

Comment:
```suggestion
    @overload
    def get(self, option: str, default: str = ...) -> str: ...
    @overload
    def get(self, option: str, default: _T) -> str | _T: ...
    @overload
    def get(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
    @overload
    def get(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
    @overload
    def __call__(self, option: str, default: str = ...) -> str: ...
    @overload
    def __call__(self, option: str, default: _T) -> str | _T: ...
    @overload
    def __call__(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
    @overload
    def __call__(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
```
Looking at the examples, `bool` is a special case that is treated differently and results in a cast that accepts both `str` and `bool`, thus allowing `default` to be `bool` as well, but in general `default` probably can't always pass through. So it might get a little complicated, but what is definitely true however, is, that without a `cast`, the default will just pass through unchanged, so the overloads without a `cast` should look different.

I am still not a big fan of `default: str | _T`, since that technically means cast should be `Callable[[str | _T], _T]` but that's too strict if `default` is actually just a `str`. So you would need yet another overload, and it would probably end up resulting in more type errors for stuff that should be legal.

But this is kind of just an inherent problem of this API, it would make a lot more sense for `default` to just return immediately, without going through `cast`.

Reply:
Same, not a fan of `default: str | _T` and you are correct about `bool` being a special case.

Another case that prompted the change was also `int`, for example `config("EMAIL_PORT", default=25, cast=int)`.  Which is in the docs, so I wanted to make sure it's a use case that's supported.

Reply:
However, I should add that I incorporated your suggested changes in the PR.